code/WorkInProgress/artemis/GeneratePlanets.dm

```diff
@@ -243,6 +243,42 @@ var/global/datum/planetManager/PLANET_LOCATIONS = new /datum/planetManager()
 	planet_area.name = name
 	region.clean_up(main_area=planet_area)
 
+	//Parallax it?
+	if(istype(generator, /datum/map_generator/snow_generator) && prob(15) )
+		var/angle = rand(110,250)
+		var/scroll_speed = rand(50, 100)
+		var/color_alpha = rand(30,60)/100
+		var/color_matrix = list(
+								1, 0, 0, color_alpha,
+								0, 1, 0, color_alpha,
+								0, 0, 1, color_alpha,
+								0, 0, 0, 1,
+								0, 0, 0, -1)
+		planet_area.area_parallax_layers = list(
+		/atom/movable/screen/parallax_layer/foreground/snow=list(color=color_matrix, scroll_speed=scroll_speed, scroll_angle=angle),
+		/atom/movable/screen/parallax_layer/foreground/snow/sparse=list(color=color_matrix, scroll_speed=scroll_speed+25, scroll_angle=angle),
+		)
+	else if(istype(generator, /datum/map_generator/desert_generator)&& prob(15) )
+		var/angle = rand(110,250)
+		var/scroll_speed = rand(75, 175)
+		var/color_alpha = rand(40,80)/100
+		var/color_matrix = list(
+								1, 0, 0, color_alpha,
+								0, 1, 0, color_alpha,
+								0, 0, 1, color_alpha,
+								0, 0, 0, 1,
+								0, 0, 0, -1)
+		planet_area.area_parallax_layers = list(
+			/atom/movable/screen/parallax_layer/foreground/dust=list(color=color_matrix, scroll_speed=scroll_speed, scroll_angle=angle),
+			/atom/movable/screen/parallax_layer/foreground/dust/sparse=list(color=color_matrix, scroll_speed=scroll_speed*1.5, scroll_angle=angle)
+		)
+
+	// Occlude overlays on edges
+	if(planet_area.area_parallax_layers)
+		for(var/turf/cordon/CT in planet_area)
+			new/obj/foreground_parallax_occlusion(CT)
```

Comment:
I would recommend replacing `/obj/foreground_parallax_occlusion` with an `UpdateOverlays()` call, so to remain consistant with existing occlusion overlays and prevent edge case bugs with using object overlays.

Using a key other than `"foreground_parallax_occlusion_overlay"` would be advisable, so to prevent undesirable interactions with the existing occlusion overlays.

Reply:
This is specifically to do deal with the the fact that cordon turfs are invisible.

Reply:
The `RESET_ALPHA` or `KEEP_APART` appearance flags may also be used to resolve that.

Reply:
Ooooo

Reply:
Seems to still be an issue due to invisibility. 🙁

Reply:
Though having the overlay present should help standard turfs expanding their overlay to the edge and not just having a strip down the center.